projects/RabbitMQ.Client/client/impl/SocketFactory.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using RabbitMQ.Client.Exceptions;
+
+namespace RabbitMQ.Client.client.impl
+{
+    internal static class SocketFactory
```

Comment:
I moved these method out here to free up the SocketHandler with this "unrelated" code.